### ISSUE TITLE
How to send a PM?

### ISSUE THREAD

**Issue by Zeks** [yourWaifu/sleepy-discord#132]
How do I reply to a user in PM?

**Comment by Zeks** [comment]
Doesn't work with c++17 because I can't compile the bot if I set c++17 in cmakelists and otherwise constructor for Channel from basic_string_view is not defined (lib uses nonstd). 
![image](https://user-images.githubusercontent.com/1160852/50574473-4e690000-0dfa-11e9-9c48-cfe91592331a.png)

**Comment by yourWaifu** [comment]
I don't see how c++17 brakes the above code

**Comment by Zeks** [comment]
My IDE sees this code as
![image](https://user-images.githubusercontent.com/1160852/50574659-93dafc80-0dfd-11e9-8b82-e2b4ebc574e3.png)

Is this correct? Maybe c++17 triggers incorrect ifdef branch here?

**Comment by yourWaifu** [comment]
It's correct, c++17 should have return type detection do it should use the top one.

**Comment by Zeks** [comment]
![image](https://user-images.githubusercontent.com/1160852/50574679-2d0a1300-0dfe-11e9-97fd-2455d779e16e.png)
full error

**Comment by Zeks** [comment]
```
set_target_properties(sleepy-discord PROPERTIES CXX_STANDARD 17)
set_target_properties(sleepy-discord PROPERTIES CXX_STANDARD_REQUIRED ON)
target_compile_features(sleepy-discord PUBLIC cxx_std_17)
```

this is what I changed in cmakelists.txt

**Comment by Zeks** [comment]
For the reference: I am compiling with gcc 8.2

**Comment by Zeks** [comment]
![image](https://user-images.githubusercontent.com/1160852/50577953-f2c36480-0e43-11e9-9ae0-5f2f9341c067.png)
It looks like repeatedly opening private communications channel somehow kills the bot's poll cycle(my assumption). It doesn't crash but after a time of running(like 5-15 minutes) the code like above it stops picking up new messages at all. 

With the private channel line commented out, this doesn't happen

**Comment by Zeks** [comment]
across 2 channels the bot is sitting at it receives maybe 60-200 messages in that interval of time

**Comment by yourWaifu** [comment]
I'm not sure how that's going to work. There's only one onMessage event and the library calls the function on the same thread that processed the event. I guess what you can do is create a new thread or use the schedule function to have it asynchronicity do it later and all of them should be done at some point.

As for me, I do plan on better async HTTP request so that it doesn't take up cycles on a thread that should be used to processing on coming data from Discord instead of waiting for Discord's response.

**Comment by Zeks** [comment]
This doesn't explain why the bot stops picking up new messages and "goes offline" on the discord. Unfortunately, if it's not addrerssed I am going to have to move to something else.

It's not an asynhronicity problem, it's "the bot is straight up broken and it's core loop dies" problem.

**Comment by Zeks** [comment]
It doesn't stop on createDirectMessageChannel  or anything, it just stops receiving new onMessage event's without an identifiable reason

**Comment by yourWaifu** [comment]
I do however think it might be an asynchronicity problem since Discord expects a response or some action back from some actions at specified times. Missing these will cause a Disconnect. If this is the case, it will say something like ``close`` in the console with a number and a reason.

**Comment by Zeks** [comment]
Nope, there are no console messages, I did see those when I hit the limits, but in this situation "pings" from the line of the code in the post above just stop coming without any kind of diagnostic

**Comment by Zeks** [comment]
oops, sclosed accidentallly

**Comment by Zeks** [comment]
I will try to make sure I open  a PM channel only for actual responses tho if that might be the case. It's simple enough fix so why not

**Comment by SapphireExile** [comment]
Having an issue related to this. If I attempt to send a DM to a channel that's already been created, I get a "vector subscript out of range" on channel creation. Works fine for new channels though.
